rosidl_generator_c/src/string_functions.c

```diff
@@ -84,14 +82,15 @@ rosidl_generator_c__String__assignn(
   if (n == SIZE_MAX) {
     return false;
   }
-  char * data = (char *)malloc(n + 1);
-  if (!data) {
+  str->data = realloc(str->data, n + 1);
+  if (!str->data) {
+    return false;
+  }
+  str->data = strncpy(str->data, value, n);
+  if (!str->data) {
```

Comment:
In the case it fails here is the `str->data` still guaranteed to be in a consistent state? If it was shrunk it might not be null terminated anymore.